translation-service-etranslation/src/main/java/eu/europeana/api/translation/service/etranslation/RedisMessageListener.java

```diff
@@ -26,15 +33,20 @@ public void onMessage(Message message, byte[] pattern) {
           this.message=messageBody;
         }
         else {
-          /* 
-           * the received message is treated as a json object and we need some adjustments for the escaped characters
-           * (this only applies if we get the translated text from the translated-text field in the eTransl callback,
-           * which happens if we send the text to be translated in the textToTranslate request param)
-           */
-          //remove double quotes at the beginning and at the end of the response, from some reason they are duplicated
-          String messageRemDuplQuotes = messageBody.replaceAll("^\"|\"$", "");
-          //replace a double backslash with a single backslash
-          this.message = messageRemDuplQuotes.replace("\\n", "\n");
+          if(messageAsDocument) {
+            this.message = messageBody;
+          }
+          else {
+            /* 
+             * the received message is treated as a json object and we need some adjustments for the escaped characters
+             * (this only applies if we get the translated text from the translated-text field in the eTransl callback,
+             * which happens if we send the text to be translated in the textToTranslate request param)
+             */
+            //remove double quotes at the beginning and at the end of the response, from some reason they are duplicated
+            String messageRemDuplQuotes = messageBody.replaceAll("^\"|\"$", "");
```

Comment:
this doesn't look right, we need to investigate the issue and eventually report a bug to eTRansaltion